newrelic/structures_newrelic_infra_alert_condition.go

```diff
@@ -171,7 +171,7 @@ func validateAttributesForType(c *alerts.InfrastructureCondition) error {
 		if c.Critical.Function != "" {
 			return fmt.Errorf("time_function is not supported by condition type %s", c.Type)
 		}
-		if *c.Critical.Value >= 0.0 {
+		if *c.Critical.Value != 0.0 {
```

Comment:
This seems like we should also check for nil pointer here as well.

Reply:
validateAttributesForType is only called in expandInfraAlertCondition where condition.Critical is filled by expandInfraAlertThreshold(d.Get("critical")) where d.Get is the terraform sdk function that given the schema.ResourceData returns always the default value if it does not exists in the configuration as explained in the doc:

```
// If the key does exist in the schema but doesn't exist in the configuration,
// then the default value for that type will be returned. For strings, this is
// "", for numbers it is 0, etc.
```

so, in expandInfraAlertThreshold, ```val, ok := rah["value"];``` ```ok``` will always be true and so alertInfraThreshold.Value would always be a valid pointer.

But yes, we could add a check for nil pointer there to be extra sure and future proof.

Reply:
Ah I see what you mean.  Thank you for making the change though.  I just had nil pointer checks on the brain I suppose. 

The CLA looks good, thank you, and I'll ensure the tests run locally before merge.